### PR TITLE
ddl: migrate test-infra to testify for db_cache_test.go

### LINKED ISSUE #30086: migrate test-infra to testify for ddl/db_cache_test.go

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #30086

Problem Summary:

### What is changed and how it works?

migrate test-infra to testify for db_cache_test.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@tisonkun @tangenta @wjhuang2016 PTAL

File: ddl/db_cache_serial_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/pingcap/tidb/errno"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/parser/terror"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/stretchr/testify/require"
+)
+
+// test alter table cache
```
Comment (tisonkun) [inline ddl/db_cache_serial_test.go:32]: redundant comments

Comment (tisonkun) [inline ddl/db_cache_serial_test.go:32]: redundant comments

**Review by tisonkun — APPROVED** [review]
LGTM. One comment inline.

File: ddl/db_cache_serial_test.go (hunk 1 above)
Comment (hawkingrei) [reply ddl/db_cache_serial_test.go]: removed

Comment (hawkingrei) [reply ddl/db_cache_serial_test.go]: removed

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**